content/40.units_and_quantities.md

```diff
@@ -1,269 +1,642 @@
+perform dimensional analysis on the simulation data, adding additional opportunities for catching
+errors in a data processing pipeline. In addition, it becomes straightforward to convert data from
+one unit system to another.
+
+In `yt` 4.0 we handle units in a an automatic fashion, with a symbolic units system. Originally part
+of `yt` itself since version 3.0, the unit system has now been split off into a standalone package,
+`unyt`. `unyt` leverages the symbolic math library `sympy` in combination with NumPy. In what
+follows, we will refer to `unyt` when describing the unit implementation, unless we are describing a
+`yt`-specific extension.
+
+Instead of returning a NumPy `ndarray` when users query `yt` data objects for fields, we return a
+`unyt_array`, a subclass of `ndarray`. `unyt_array` preserves `ndarray`'s array operations,
+including deep and shallow copies, broadcasting, views, and mathematical operations. Augmenting
+`ndarray`, `unyt_array` attaches unit metadata to the array data, enabling runtime checking of unit
+consistency in arithmetic operations between `unyt_array` instances, and making it trivial to
+compose new units using algebraic operations.
+
+As a trivial example, when one queries a data object (here given the generic name `dd`) for the
+density field, we get back a `unyt_array`, including both the simulation data for the density field,
+and the units of the density field, in this case $\rm{g}/\rm{cm}^3$:
 
 ```python
->>> dd[`density'] 
-YTArray([4.92e-31, 4.94e-31, 4.93e-31, ...,
-         1.12e-25, 1.59e-25, 1.09e-24]) g/cm**3
+>>> dd['density'] 
+unyt_array([4.92e-31, 4.94e-31, 4.93e-31, ...,
+            1.12e-25, 1.59e-25, 1.09e-24], 'g/cm**3')
 ```
 
-One of the nicest aspects of this new unit system is that the symbolic algebra for unitful operations is performed automatically by sympy:
+One of the nicest aspects of the unit system is that the symbolic algebra for unitful operations is performed automatically by `sympy`:
 
 ```python
->>> print(dd[`cell_mass']/dd[`cell_volume'])
+>>> print(dd['mass']/dd['cell_volume'])
   [4.92e-31 4.94e-31 4.93e-31 ... 
    1.12e-25 1.59e-25 1.09e-24] g/cm**3
 ```
 
-YTArray is primarily useful for attaching units to NumPy `ndarray` instances.
-For scalar data, we have created the new `YTQuantity` class.
-`YTQuantity` is a subclass of `YTArray` with the requirement that the "array data" associated with the instance be limited to one element.
-`YTQuantity` is primarily useful for physical constants and ensures that the units are propagated correctly when composing quantities from arrays, physical constants, and unitless scalars:
+`unyt_array` is primarily useful for attaching units to NumPy `ndarray` instances. For scalar data,
+we have created the `unyt_quantity` class. `unyt_quantity` is a subclass of `unyt_array` with the
+requirement that the "array data" associated with the instance be limited to one element.
```

Comment:
I'm having trouble coming up with a better phrasing but this is itching me: `unyt_quantity` is in fact slightly more constrained than this, because it can't have `ndim>0` (but it's possible to construct a one-element array in arbitrary dimensions)

Reply:
Tried to rephrase it a bit to be more precise